### PR TITLE
Remove IfConfigDecl from the AST

### PR DESCRIPTION
**Pull request by DougGregor**
The swift-syntax tree retains information about the parsed #if regions. Drop it from the semantic AST.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci please test source compatibility

File: lib/Parse/ParseIfConfig.cpp [hunk 1]
```diff
@@ -972,34 +972,25 @@ Result Parser::parseIfConfigRaw(
   return finish(EndLoc, HadMissingEnd);
 }
 
-/// Parse and populate a #if ... #endif directive.
```
Comment (hamishknight) [inline lib/Parse/ParseIfConfig.cpp:975]: Is this intentional?

**Review by hamishknight — APPROVED** [review]

**Comment by DougGregor** [comment]
https://github.com/swiftlang/llvm-project/pull/9291

@swift-ci please test

File: lib/Parse/ParseIfConfig.cpp (hunk 1 above)
Comment (DougGregor) [reply lib/Parse/ParseIfConfig.cpp:975]: ... nope, accident of refactoring. Thank you!

**Comment by slavapestov** [comment]
Beautiful!

File: lib/Parse/ParseIfConfig.cpp (hunk 1 above)
Comment (DougGregor) [reply lib/Parse/ParseIfConfig.cpp:975]: Oh, this is the .cpp file. This function is documented in the header, which is good enough. We don't need to duplicate the comment here.

**Pull request merged**